src/main/java/com/github/susom/database/VertxUtil.java

```diff
@@ -99,15 +99,15 @@
    * Equivalent to {@link Vertx#executeBlocking(Handler, Handler)},
    * but preserves the {@link MDC} correctly.
    */
-  public static <T> void executeBlocking(Vertx vertx, Handler<Future<T>> future, Handler<AsyncResult<T>> handler) {
+  public static <T> void executeBlocking(Vertx vertx, Handler<Promise<T>> future, Handler<AsyncResult<T>> handler) {
```

Comment:
Change the parameter name for `future`?  (and similarly below)

Reply:
Updated